Chore: Set up the frontend

laraduerlich opened the pull request:
close #1 

- [x] chore: set up the ordner structure
- [x] feat: create LandingPage
- [x] feat: create DashboardPage
- [x] feat: create SummaryPage
- [x] feat: create LinkPage
- [x] feat: create a header
- [x] feat: create a footer
- [x] feat: create the routes in the App.tsx
- [x] feat: create the CSS basic structure 


<img width="1787" alt="image" src="https://github.com/user-attachments/assets/9e6064a9-41cc-4897-b1d6-5f3238be7277" />

Commit:
feat: add pages and routes in the app

Commit:
feat: add tailwind config and basic css structure

mpagels commented on frontend/src/page/DashboardPage.tsx:
```diff
@@ -0,0 +1,8 @@
+        <>
+        </>
+    )
+}
```
It is bad practice to implement empty components.
In a real world scenario, it can and will happen, that you have to do other things next and then these empty components staying in the project for half a year. If these components don't doing anything. Don't put it in your production code.

mpagels commented on frontend/src/page/LandingPage.tsx:
```diff
@@ -0,0 +1,9 @@
+
+        </>
+    )
+}
```
Same as above with the empty components. ;-)

mpagels commented on frontend/src/page/SummaryPage.tsx:
```diff
@@ -0,0 +1,8 @@
+        <>
+        </>
+    )
+}
```
And here an empty component

mpagels reviewed (approved):
This pull request is too big and doing too much. (Bad practice)

This pull request
- changes the css in the project
- implements react-router
- implements a navigation
- and implements some components

For a `chore` task is it way too much.
Try to make one pull request for every big step.
Why you're asking? Here is the explanation:

Let's say you implement the pull request as it is, with all the changes you're proposing.
Some work and improvement later, you want to switch from tailwind css to mui-react (another css library for react). If you had implemented your previous steps, one pull request by one, you could just revert that "css" commit, and can implement the new css library. In your current state of committing your work, reverting the pull request commit, you would revert everything. The css, react-router, navigation, the components. That would suck.
And as you can not just revert the css commit, you now have to dig in, and find all the files from tailwind css to delete manually.

That said, I highly recommend using smaller pull request, that doing one thing.
This has several positive effects. One I explained above, the other one is that smaller pull request are likely to review. A fellow programmer would, review just 40 line of changed code, then 800 lines of code.

For now: I will approve this pull request, because it looks good, but I would be happy if the next pull request would be smaller and implements the good practice I wrote down.

If you have any questions, pls don't hesitate to contact me via slack or zoom and we can talk about it. :-) 

At last: Very good work here, setting the frontend up. I like to see just frontend code added to the project. Keep it up!!! 

PS: I left some notes on the some line of codes.

Commit:
feat: add headers in the pages

laraduerlich commented:
> This pull request is too big and doing too much. (Bad practice)
> 
> This pull request
> 
> * changes the css in the project
> * implements react-router
> * implements a navigation
> * and implements some components
> 
> For a `chore` task is it way too much. Try to make one pull request for every big step. Why you're asking? Here is the explanation:
> 
> Let's say you implement the pull request as it is, with all the changes you're proposing. Some work and improvement later, you want to switch from tailwind css to mui-react (another css library for react). If you had implemented your previous steps, one pull request by one, you could just revert that "css" commit, and can implement the new css library. In your current state of committing your work, reverting the pull request commit, you would revert everything. The css, react-router, navigation, the components. That would suck. And as you can not just revert the css commit, you now have to dig in, and find all the files from tailwind css to delete manually.
> 
> That said, I highly recommend using smaller pull request, that doing one thing. This has several positive effects. One I explained above, the other one is that smaller pull request are likely to review. A fellow programmer would, review just 40 line of changed code, then 800 lines of code.
> 
> For now: I will approve this pull request, because it looks good, but I would be happy if the next pull request would be smaller and implements the good practice I wrote down.
> 
> If you have any questions, pls don't hesitate to contact me via slack or zoom and we can talk about it. :-)
> 
> At last: Very good work here, setting the frontend up. I like to see just frontend code added to the project. Keep it up!!!
> 
> PS: I left some notes on the some line of codes.

Thank you, I will write the next PR a little bit smaller :) I also fixed the few issues and added headers in the pages. Could you please approve it?
